layerfetcher/layers.go

```diff
@@ -305,6 +305,28 @@ func (l *Logic) PollLayerContent(ctx context.Context, layerID types.LayerID) cha
 	return resChannel
 }
 
+// registerLayerHashes registers provided hashes with provided peer.
+func (l *Logic) registerLayerHashes(peer p2p.Peer, blocks *layerData) {
+	if blocks == nil {
+		return
+	}
+	var layerHashes []types.Hash32
+	for _, ballotID := range blocks.Ballots {
+		layerHashes = append(layerHashes, ballotID.AsHash32())
+	}
+	for _, blkID := range blocks.Blocks {
+		layerHashes = append(layerHashes, blkID.AsHash32())
+	}
```

Comment:
i'd also add HareOutput just in case it's not included in layerData.Blocks.

Reply:
done